bigtable-dataflow-parent/bigtable-beam-import/README.md

```diff
@@ -1,44 +1,204 @@
+### Download the jars
+Download [the import/export jars](http://search.maven.org/remotecontent?filepath=com/google/cloud/bigtable/bigtable-beam-import/1.14.1/bigtable-beam-import-1.14.1-shaded.jar), which is an aggregation of all required jars.
+
+### Build the jars yourself
+
+Go to the top level directory and build the repo
+then return to this sub directory.
+
+```
+cd ../../
+mvn clean install -DskipTests=true
+cd bigtable-dataflow-parent/bigtable-beam-import
+```
 
-Please pay attention to the Cluster CPU usage and adjust the number of Dataflow workers accordingly.
 
-## Export
+## Exporting data
+
+This folder contains pipelines to help export data via snapshots or into sequence files
+and can be used for HBase or Bigtable. If you're migrating your data from HBase to Bigtable
+using snapshots is the preferred method. 
+
+### Exporting snapshots from HBase
+
+```
+SNAPSHOT_NAME=your-snapshot-name
+```
+
+```
+hbase org.apache.hadoop.hbase.snapshot.ExportSnapshot -snapshot $SNAPSHOT_NAME \
+    -copy-to /integration-test/data -mappers 16
+```
+
+### Exporting sequence files from HBase
+
+On your HDFS set the environment variables.
+
+```
+TABLENAME="my-new-table"
+EXPORTDIR=/usr/[USERNAME]/hbase-${TABLENAME}-export
+hadoop fs -mkdir -p ${EXPORTDIR}
+MAXVERSIONS=2147483647
+```
+
+Then in your HBase home directory run the export commands. 
+
+```
+cd $HBASE_HOME
+bin/hbase org.apache.hadoop.hbase.mapreduce.Export $TABLENAME \
+    /user/hbase-$TABLENAME \
+    -export $MAXVERSIONS
+bin/hbase org.apache.hadoop.hbase.mapreduce.Export \
+    -Dmapred.output.compress=true \
+    -Dmapred.output.compression.codec=org.apache.hadoop.io.compress.GzipCodec \
+    -DRAW_SCAN=true \
+    -Dhbase.client.scanner.caching=100 \
+    -Dmapred.map.tasks.speculative.execution=false \
+    -Dmapred.reduce.tasks.speculative.execution=false \
+    $TABLENAME $EXPORTDIR $MAXVERSIONS
+
+```
+
+### Exporting snapshots from Bigtable
+
+Exporting HBase snapshots from Bigtable is not supported.
+
+### Exporting sequence files from Bigtable
+
+Set the environment variables.
+
+```
+PROJECT_ID=your-project-id
+INSTANCE_ID=your-instance-id
+CLUSTER_NUM_NODES=3
+TABLE_NAME=your-table-name
 
-On the command line:
+BUCKET_NAME=gs://bucket-name
+```
 
+Run the export.
 ```
 java -jar bigtable-beam-import-1.14.1-shaded.jar export \
     --runner=dataflow \
-    --project=[your_project_id] \
-    --bigtableInstanceId=[your_instance_id] \
-    --bigtableTableId=[your_table_id] \
-    --destinationPath=gs://[bucket_name]/[export_directory]/ \
-    --tempLocation=gs://[bucket_name]/[temp_work_directory]/ \
-    --maxNumWorkers=[10x number of nodes] \
+    --project=$PROJECT_ID \
+    --bigtableInstanceId=$INSTANCE_ID \
+    --bigtableTableId=$TABLE_NAME \
+    --destinationPath=$BUCKET_NAME/hbase_export/ \
+    --tempLocation=$BUCKET_NAME/hbase_temp]/ \
+    --maxNumWorkers=$(expr 10 \* $CLUSTER_NUM_NODES) \
```

Comment:
is 10x the right number here? I've typically seen 3x

Reply:
10x is much higher. I agree 3x works much better.

Reply:
the `expr 10 \*` just does the arithmetic inline, so you only need to provide the number of nodes for your cluster